Dev tls

jakewatson-bristol opened the pull request:
generates a self signed CA for cert manager to use to generate tls certificates

I used crds2pulumi to make a cert manager module but if we think this isn't needed I'd imagine it can be swapped out for some generic pulumi way to deploy unknown manifests

https://www.pulumi.com/blog/introducing-crd2pulumi/

jakewatson-bristol commented:
looks like I've accidentally include a small part of the OKE stuff I'm working on but hopefully not an issue (as I think it's not part of it's own commit)

jakewatson-bristol commented:
relavent links:
https://cert-manager.io/docs/configuration/ca/
https://www.pulumi.com/blog/introducing-crd2pulumi/
https://cryptography.io/en/latest/x509/tutorial/#creating-a-self-signed-certificate

jakewatson-bristol commented:
Ah didn't realise I had a missed that their was already a pumumi cert manager package I'll move over to using that to avoid the big import from crd2pumumi

and yes sorry the OKE stuff keeps slipping in by mistake bad git usage on my end

jakewatson-bristol commented on infra/fridge/requirements.txt:
```diff
@@ -1,3 +1,5 @@
 pulumi-azuread~=6.0
 pulumi-kubernetes~=4.0
 pulumi~=3.0
+requests[socks]
```
slipped in from the the SSH proxy work I was doing it's pretty much the only change that is required to allow `pulumi up` to work VIA a SSH socks tunnel

jakewatson-bristol commented on infra/fridge/__main__.py:
```diff
@@ -136,6 +145,109 @@ def patch_namespace(name: str, pss: PodSecurityStandard) -> NamespacePatch:
+    # Generate private key
+    private_key = rsa.generate_private_key(
+        public_exponent=65537,
+        key_size=2048,
+    )
+
+    # Build subject and issuer (self-signed CA)
+    subject = issuer = x509.Name(
+        [
+            x509.NameAttribute(NameOID.COUNTRY_NAME, country),
+            x509.NameAttribute(NameOID.LOCALITY_NAME, locality),
+            x509.NameAttribute(NameOID.ORGANIZATION_NAME, organization),
+            x509.NameAttribute(NameOID.COMMON_NAME, commonName),
+            x509.NameAttribute(NameOID.EMAIL_ADDRESS, emailAddress),
+        ]
+    )
+
+    cert = (
+        x509.CertificateBuilder()
+        .subject_name(subject)
+        .issuer_name(issuer)
+        .public_key(private_key.public_key())
+        .serial_number(x509.random_serial_number())
+        .not_valid_before(validity_start)
+        .not_valid_after(validity_end)
+        .add_extension(
+            x509.BasicConstraints(ca=True, path_length=None),
+            critical=True,
+        )
+        .sign(private_key, hashes.SHA256())
+    )
+
+    # Return PEM-encoded private key and certificate
+    private_key_pem = private_key.private_bytes(
+        encoding=serialization.Encoding.PEM,
+        format=serialization.PrivateFormat.TraditionalOpenSSL,
+        encryption_algorithm=serialization.NoEncryption(),
+    )
+    cert_pem = cert.public_bytes(serialization.Encoding.PEM)
+    return private_key_pem, cert_pem
```
I think Utils might be a better name, when i see scripts i think of standalone units i can just run my calling the file

so I'll move the cert generation into infra/fridge/utils/cert-generator.py

and get rid of the crd2pulumi stuff in favour of the cert manager pulumi package

craddm replied:
You can use Pulumi's [CustomResource](https://www.pulumi.com/registry/packages/kubernetes/api-docs/apiextensions/customresource/) to create Issuers

There's an example on their Github repo: https://github.com/pulumi/pulumi-kubernetes-cert-manager/blob/master/examples/simple-cert-manager-py/__main__.py

(Incidentally, I think you may be able to get `cert-manager` to [generate the certificate for you](https://cert-manager.io/docs/configuration/selfsigned/) but am happy to do it whichever way you like)

(nb, I only realised you could use `CustomResource` for this while reviewing this PR, so I'll be using it for some other things soon...!)

jakewatson-bristol replied:
Nice that has worked using cert manager to generate the certificate
<img width="1588" height="594" alt="image" src="https://github.com/user-attachments/assets/df7ee019-7416-4a3b-a5d0-8b943e0da92f" />

much simpler this way, I've given the dev CA certificate a wildcard because it's easier for dev

jakewatson-bristol commented:
okay this should be better

Removed crd2pulumi code
Removed cert generator code

add 2 Cluster Issuer CustomResourse
  selfsigned - generates the CA cert for the issuer that is used
 
 add 1 Certificate CustomResourse
   used as CA cert signed by self signed issuer
<img width="1588" height="594" alt="image" src="https://github.com/user-attachments/assets/0e4bb111-76dc-4501-a630-b3628bdb0187" />

jakewatson-bristol commented:
> Nice! Still a couple of vestigial OKE bits. The merge conflict is a pretty simple one - happy to sort that out before merge

Forgot to check the enums, thanks for letting me know
